modules/core/nuxeo-core-event/src/main/java/org/nuxeo/ecm/core/work/WorkManagerImpl.java

```diff
@@ -426,41 +427,34 @@ public void init() {
             for (WorkQueueDescriptor descriptor : descriptors) {
                 initializeQueue(descriptor);
             }
+        }
+    }
 
-            Framework.getRuntime().getComponentManager().addListener(new ComponentManager.Listener() {
-                @Override
-                public void beforeStop(ComponentManager mgr, boolean isStandby) {
-                    List<WorkQueueDescriptor> descriptors = getDescriptors(QUEUES_EP);
-                    for (WorkQueueDescriptor descriptor : descriptors) {
-                        deactivateQueue(descriptor);
-                    }
-                    try {
-                        if (!shutdown(10, TimeUnit.SECONDS)) {
-                            log.error("Some processors are still active");
-                        }
-                    } catch (InterruptedException e) {
-                        Thread.currentThread().interrupt();
-                        throw new NuxeoException("Interrupted while stopping work manager thread pools", e);
-                    }
-                }
-
-                @Override
-                public void afterStart(ComponentManager mgr, boolean isResume) {
-                    if (isProcessingDisabled()) {
-                        log.warn("WorkManager processing has been disabled on this node");
-                        return;
-                    }
-                    List<WorkQueueDescriptor> descriptors = getDescriptors(QUEUES_EP);
-                    for (WorkQueueDescriptor descriptor : descriptors) {
-                        activateQueue(descriptor);
-                    }
-                }
+    @Override
+    public void afterStart(ComponentManager mgr, boolean isResume) {
```

Comment:
The existing naming for `ComponentManager.Listener` methods gets a bit confusing when we have this method inside a `Component`, because now we have a `start` method that is called when the component is started, but `afterStart` is after the _whole runtime_ has started. Could you look at renaming/aliasing things so that here we can have a method name `afterRuntimeStart` (for instance).

Reply:
This was just a suggestion to improve listeners declaration, we can review the API indeed but i'll create a jira first if we're happy with this way of registering listeners, to address the methods confusion.